trl/trainer/judges.py

```diff
@@ -303,3 +476,31 @@ def get_rank(prompt, candidates):
 
         # Return the ranks
         return ranks
+
+
+class MixtureOfConstraintJudges(BaseConstraintJudge):
+    """
+    Unify the decision of multiple BaseConstraintJudge.
+
+    This class returns False if it fails on any of the constraint judges (ie a judge returns 0 or -1) and returns True otherwise.
+
+    It is an implementation of the Mixture of Judges as described in the [CGPO paper](https://huggingface.co/papers/2409.20370)
+
+    Args:
+        judges (`List[BaseConstraintJudge]`): A list of [`BaseConstraintJudge`].
+    """
+
+    def __init__(self, judges: List[BaseConstraintJudge]):
+        self.judges = judges
+
+    def judge(
+        self, prompts: List[str], completions: List[str], gold_answers: List[str] = None, shuffle_order: bool = True
```

Comment:
can the gold_answer be none? If not I'd make it required arg.

Reply:
Yes, the SafetyJudge of the paper do not use a gold answer for example